app/views/katello/api/v2/subscriptions/base.json.rabl

```diff
@@ -16,7 +16,7 @@ attributes :name => :product_name
 attributes :unmapped_guest
 attributes :virt_only
 attributes :virt_who
-attributes :upstream? => :upstream
+attributes :upstream_pool_id
```

Comment:
We normally have a policy that we can only reach out to the backend for attributes on 'show' requests, not index (since it would trigger a call for each item in the list).  If we want to return this as part of the index call, we'll need to pull it into the db.

Reply:
I agree with that.

Here's another approach:

1. Reinstate the 'upstream?' attribute returned by this API
2. Remove upstream_pool_id which I added to the index
3. From the API side, instead of taking (upstream) pool_ids as a parameter (which requires we know the upstream pool ID) in order to get the quantity available, send in local subscription ids and perform the lookup on the backend
4. avoid import of candlepin data (profit)

We kind of do similar already for updating the upstream entitlement qty for example - taking the local sub ID as input, and lookup the upstream_entitlement_id in the action

Reply:
@tstrachota ok, the approach here will be changing. going to work on it now